src/MSBuild.UnitTests/Microsoft.Build.CommandLine.UnitTests.csproj

```diff
@@ -42,6 +42,7 @@
     <Compile Include="..\Shared\UnitTests\MockEngine.cs" />
     <Compile Include="..\Shared\UnitTests\MockLogger.cs" />
     <Compile Include="..\Shared\UnitTests\ObjectModelHelpers.cs" />
+    <Compile Include="..\UnitTests.Shared\EnvironmentProvider.cs" />
```

Comment:
nit: just see there is `Shared\UnitTests` and `UnitTests.Shared`: what is the difference here?

Reply:
Nice catch!

For historical reasons, there is a bunch of code in MSBuild that is compiled into multiple assemblies (sometimes with slightly different ifdefs). That includes test assemblies. However, the reasons for doing this in non-test assemblies (they aren't great reasons but they are reasons) don't really apply for test, so we introduced a shared assembly for test stuff to build that sort of thing once--but didn't move everything.

Reply:
Actually it looks like shortly after introducing that assembly, we dropped all references to it (in ~2017).

It's still a good idea.